### PR TITLE
Migrate the deploy codebase aside from release code

### PR DESCRIPTION
**Pull request by inlined**
Migrates:
* ContainerCleaner
* Deploy
* Prepare
* Pricing
* Prompts

These changes are all safe because backend now has redundant information in endpoints and functions.

### CODE REVIEW

File: src/deploy/functions/prompts.ts [hunk 1]
```diff
@@ -19,33 +19,30 @@ import { Options } from "../../options";
...
-  want: backend.FunctionSpec[],
-  have: backend.FunctionSpec[]
+  want: backend.Backend,
+  have: backend.Backend
 ): Promise<void> {
   // Collect all the functions that have a retry policy
-  const retryFunctions = want.filter((fn) => {
-    return backend.isEventTrigger(fn.trigger) && fn.trigger.retry;
+  const retryEndpoints = backend.allEndpoints(want).filter((fn) => {
+    return backend.isEventTriggered(fn) && fn.eventTrigger.retry;
```
Comment (taeold) [inline src/deploy/functions/prompts.ts:27]:
```suggestion
  const retryEndpoints = backend.someEndpoint(wantBackend, (e) => {
    return backend.isEventTriggered(e) && e.eventTrigger.retry;
```

File: src/deploy/functions/prompts.ts [hunk 2]
```diff
@@ -133,20 +130,20 @@ export async function promptForFunctionDeletion(
...
-  have: backend.FunctionSpec[]
+  want: backend.Backend,
+  have: backend.Backend
 ): Promise<void> {
   if (options.force) {
     return;
   }
 
-  const increasesCost = want.some((wantFn) => {
+  const increasesCost = backend.allEndpoints(want).some((wantFn) => {
```
Comment (taeold) [inline src/deploy/functions/prompts.ts:140]: nit* rename `wantFn` to `wantE`?

File: src/deploy/functions/prompts.ts [hunk 3]
```diff
@@ -174,7 +171,8 @@ export async function promptForMinInstances(
   // Considerations for future versions:
   // Group Tier 1 and Tier 2 regions
   // Add Tier 1 or Tier 2 annotations to functionLines
-  const functionLines = want
+  const functionLines = backend
+    .allEndpoints(want)
     .filter((fn) => fn.minInstances)
```
Comment (taeold) [inline src/deploy/functions/prompts.ts:176]: nit* `fn` => `e` (here and onwards)

**Review by taeold — APPROVED** [review]

File: src/deploy/functions/backend.ts [hunk 4]
```diff
@@ -567,9 +567,45 @@ export function allEndpoints(backend: Backend): Endpoint[] {
...
+    }
+    filtered.endpoints[endpoint.region] = filtered.endpoints[endpoint.region] || {};
+    filtered.endpoints[endpoint.region][endpoint.id] = endpoint;
+  }
+  return filtered;
+}
+
+/** A helper utility for flattening all endpoints in a region since typing is a bit wonky. */
+export function regionalEndpoints(backend: Backend, region: string): Endpoint[] {
+  return Object.values<Endpoint>(backend.endpoints[region]);
```
[the remark this answers is not in the document]
Comment (inlined) [reply src/deploy/functions/backend.ts:603]: Great suggestion. Adding manually.

File: src/deploy/functions/prompts.ts (hunk 1 above)
Comment (inlined) [reply src/deploy/functions/prompts.ts:27]: I'll change the parameter name, but `allEndpoints.filter` is intentional. we're getting a list, not a boolean.

File: src/deploy/functions/prompts.ts (hunk 2 above)
Comment (inlined) [reply src/deploy/functions/prompts.ts:140]: Sure

File: src/deploy/functions/prompts.ts (hunk 3 above)
Comment (inlined) [reply src/deploy/functions/prompts.ts:176]: It's actually against our prettier according to my plugin, but I like it better. I'll manually change and investigate later

**Pull request merged**